naming/src/main/java/org/jboss/as/naming/ValueManagedReference.java

```diff
@@ -24,14 +24,31 @@
 import org.jboss.msc.value.Value;
 
 /**
- * A ManagedReference that simply holds a value'
+ * A ManagedReference that simply holds a value.
  *
  * @author Stuart Douglas
+ * @author <a href="mailto:ropalka@redhat.com">Richard Opalka</a>
  */
 public class ValueManagedReference implements ManagedReference {
-    private final Value<Object> value;
+    private final Object value;
 
-    public ValueManagedReference(Value<Object> value) {
+    /**
+     * Construct a new instance.
+     *
+     * @param value the value to wrap
+     * @deprecated use {#{@link ValueManagedReference(Object)}} instead. This constructor will be removed in the future.
+     */
+    @Deprecated
+    public ValueManagedReference(final Value<Object> value) {
+        this(value != null ? value.getValue() : null);
```

Comment:
This changes the semantic by reading the value at this point instead of in getInstance().

I envisioned the 'real' constructor would take and cache a supplier, which getInstance would use, and this and the public ValueManagedReference(final Object value) constructors would create a supplier.

Reply:
Now I see what you had in mind @bstansberry - PR updated accordingly.